src/main/java/org/reactome/addlinks/dataretrieval/pharos/PharosDataRetriever.java

```diff
@@ -0,0 +1,195 @@
+			{
+				// TODO: In the future, make FileRetriever in release-common-lib more generic so that it deals with HttpEntityEnclosingRequestBase
+				// and let subclasses optionally inject their own request object (HttpGet or HttpPost).
+				// But there just isn't time before Release 77 for that sort of elegant solution, so yes, there *is* some duplication of effort in this
+				// doHttpDownload method and the method it overrode.
+				HttpPost post = createPOSTRequest(timeoutInMilliseconds, numRequests);
+				int retries = this.numRetries;
+				boolean done = retries < 0;
+				while (!done)
+				{
+					String jsonResponse = "";
+					try (CloseableHttpClient client = HttpClients.createDefault();
+						CloseableHttpResponse response = client.execute(post, context); )
+					{
+						int statusCode = response.getStatusLine().getStatusCode();
+						// If status code was not 200, we should print something so that the users know that an unexpected response was received.
+						if (statusCode != HttpStatus.SC_OK)
+						{
+							reportNotOKResponse(response.getStatusLine(), statusCode);
+						}
+						else
+						{
+							jsonResponse = EntityUtils.toString(response.getEntity());
+							JSONObject jsonObj = new JSONObject(jsonResponse);
+							int numProcessed = processJSONArray(writer, jsonObj);
+							moreRecords = numProcessed >= 1;
+							logger.info("Completed request #{}", numRequests);
+							numRequests++;
+						}
+						done = true;
+					}
+					catch (ConnectTimeoutException e)
+					{
+						// we will only be retrying the connection timeouts, defined as the time required to establish a connection.
+						// we will not handle socket timeouts (inactivity that occurs after the connection has been established).
+						// we will not handle connection manager timeouts (time waiting for connection manager or connection pool).
+						e.printStackTrace();
+						logger.info("Failed due to ConnectTimeout, but will retry {} more time(s).", retries);
+						retries--;
+						done = retries < 0;
+						if (done)
+						{
+							throw new Exception("Connection timed out. Number of retries ("+this.numRetries+") exceeded. No further attempts will be made.", e);
+						}
+					}
+					catch (PharosDataException e)
+					{
+						logger.error("There was a problem with data from Pharos: " + e.getMessage(), e);
+						throw e;
+					}
+					catch (JSONException e)
+					{
+						logger.error("There was a problem with the JSON: {}", jsonResponse);
+						throw new PharosDataException("There was a problem with the JSON from Pharos: " + e.getMessage());
+					}
+					catch (HttpHostConnectException e)
+					{
+						logger.error("Could not connect to host {} ! Exception: {}", post.getURI().getHost(), e.getMessage());
+						throw e;
+					}
+					catch (IOException e)
+					{
+						logger.error("IOException caught: {}", e.getMessage());
+						throw e;
+					}
+				}
+			}
+		}
+	}
+
+	/**
+	 * Report on a not-OK (non-200) HTTP response.
```

Comment:
Maybe mention that the method does not report anything if the `statusCode` passed happens to be 200?